File: binarySearch.py
def binarySearch(array, target) -> int:
    left, right = 0, len(array)-1

    while left <= right:
        middle = (left+right)//2

        if array[middle] == target:
            return middle

        elif array[middle] < target:
            left = middle+1

        else:
            right = middle-1

    return -1

File: binarySearch.rb
def binarySearch(array,target)
    left,right=0,array.size-1

    until left > right
        middle=((left+right)/2).round
        return middle if array[middle] == target
        array[middle]<target ? left=middle+1 : right=middle-1
    end

    return -1
end
